crates/pallet-domains/src/staking.rs

```diff
@@ -147,6 +149,41 @@ pub(crate) fn do_register_operator<T: Config>(
     })
 }
 
+pub(crate) fn do_nominate_operator<T: Config>(
+    operator_id: OperatorId,
+    nominator_id: T::AccountId,
+    amount: BalanceOf<T>,
+) -> Result<(), Error> {
+    OperatorPools::<T>::try_mutate(operator_id, |maybe_operator| {
+        let operator = maybe_operator.as_mut().ok_or(Error::UnknownOperator)?;
+
+        // reserve stake balance
+        ensure!(
+            amount >= operator.minimum_nominator_stake,
+            Error::MinimumNominatorStake
+        );
+
+        ensure!(
+            T::Currency::balance_freezable(&nominator_id) >= amount,
+            Error::InsufficientBalance
+        );
+
+        T::Currency::set_freeze(
+            &T::FreezeIdentifier::staking_freeze_id(),
+            &nominator_id,
+            amount,
+        )
+        .map_err(|_| Error::BalanceFreeze)?;
+
+        operator.pending_transfers.push(PendingTransfer {
+            nominator_id,
+            transfer: Transfer::Deposit(amount),
+        });
```

Comment:
Should also update `next_total_stake`

Reply:
Not necessary as per new staking protocol